src/components/Channel/channelState.js

```diff
@@ -59,7 +59,9 @@ export const channelReducer = (state, action) => {
       if (!state.thread) return state;
       return {
         ...state,
-        threadMessages: channel.state.threads[state.thread.id],
+        threadMessages: state.thread?.id
+          ? channel.state.threads[state.thread.id]
+          : Immutable([]),
```

Comment:
I'll look at this, @vishalnarkhede made the change I think so need to dig a little.

Reply:
OK looks like we should always have a fallback for threadMessages to empty Immutable Array, it starts as that and is set to that on close so I have changed it so it will gall back to that.